### PR TITLE
Update command result message regarding large integer user input

### LINKED ISSUE #251: Parsing very large integer values
For duration, index, etc.

Should raise corresponding command exception and give appropriate command result message.

### PR DESCRIPTION
**Pull request by JeffZincatz**
Fix #251 
When a user now inputs a large invalid integer for fields such as index and duration, a corresponding command failure result message will be shown to notify the user about the mistake.

| command | field | added message |
| - | - | - |
| `edit_x` | INDEX | ..., or bigger than maximum value |
| `delete_x` | INDEX | , or bigger than maximum value |
| `add_i` `edit_i` | duration | ... and less than 1 day (1440 minutes) |

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -36,7 +36,8 @@
  */
 public class ParserUtil {
 
-    public static final String MESSAGE_INVALID_INDEX = "Index is not a non-zero unsigned integer.";
+    public static final String MESSAGE_INVALID_INDEX =
+            "Index is not a non-zero unsigned integer, or bigger than maximum value.";
```
Comment (seaweediman) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:40]: Think should specifically say what is maximum value. A typical user would be confused if you meant the number of candidates or whatever.

Comment (JeffZincatz) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:40]: True. Will make the changes accordingly.

File: src/main/java/seedu/address/logic/parser/DeleteCandidateCommandParser.java [hunk 2]
```diff
@@ -21,8 +21,8 @@ public DeleteCandidateCommand parse(String args) throws ParseException {
             Index index = ParserUtil.parseIndex(args);
             return new DeleteCandidateCommand(index);
         } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCandidateCommand.MESSAGE_USAGE), pe);
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, pe.getMessage())
+                            + "\n" + DeleteCandidateCommand.MESSAGE_USAGE, pe);
```
Comment (seaweediman) [inline src/main/java/seedu/address/logic/parser/DeleteCandidateCommandParser.java:25]: I just played around with the PR.

image

Should the Index error really be shown when no index is inputted

Comment (JeffZincatz) [reply src/main/java/seedu/address/logic/parser/DeleteCandidateCommandParser.java:25]: On the first look, the reason for this behaviour seems to be that the parsed index is an empty string (""). I feel that this case can be handled similarly to other non-numerical user inputs. Let me know what are your thoughts.

Comment (seaweediman) [reply src/main/java/seedu/address/logic/parser/DeleteCandidateCommandParser.java:25]: We had a discussion on this in the tele group

File: src/main/java/seedu/address/logic/parser/RemarkCandidateCommandParser.java [hunk 3]
```diff
@@ -26,8 +26,8 @@ public RemarkCandidateCommand parse(String args) throws ParseException {
         try {
             index = ParserUtil.parseIndex(argMultimap.getPreamble());
         } catch (IllegalValueException ive) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                    RemarkCandidateCommand.MESSAGE_USAGE), ive);
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ive.getMessage())
+                    + "\n" + RemarkCandidateCommand.MESSAGE_USAGE, ive);
```
Comment (seaweediman) [inline src/main/java/seedu/address/logic/parser/RemarkCandidateCommandParser.java:30]: Should probably remove the "\n" 

image

File: src/test/java/seedu/address/logic/parser/RemarkCandidateCommandParserTest.java [hunk 4]
```diff
@@ -25,7 +25,7 @@
     public void execute_invalidPersonIndex_throwsCommandException() {
         assertParseFailure(parser, model.getFilteredPersonList().size()
                 + VALID_REMARK_BOB, String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                RemarkCandidateCommand.MESSAGE_USAGE));
+                ParserUtil.MESSAGE_INVALID_INDEX) + "\n" + RemarkCandidateCommand.MESSAGE_USAGE);
```
Comment (seaweediman) [inline src/test/java/seedu/address/logic/parser/RemarkCandidateCommandParserTest.java:28]: Here too. Need to remove the "\n"

**Review by seaweediman — APPROVED** [review]
LGTM

**Pull request merged**